cli/command/image/tree.go

```diff
@@ -230,64 +229,76 @@ func printImageTree(dockerCLI command.Cli, view treeView) error {
 
 	// Print images
 	for _, img := range images {
-		printNames(out, columns, img, topNameColor, untaggedColor)
-		printDetails(out, columns, normalColor, img.Details)
-
-		if len(img.Children) > 0 || view.imageSpacing {
-			_, _ = fmt.Fprintln(out)
-		}
-		printChildren(out, columns, img, normalColor)
+		_, _ = fmt.Fprint(out, formatImageTree(img, columns, view.imageSpacing, normalColor, topNameColor, untaggedColor))
 		_, _ = fmt.Fprintln(out)
 	}
 
 	return nil
 }
 
-func printDetails(out *streams.Out, headers []imgColumn, defaultColor aec.ANSI, details imageDetails) {
+func formatImageTree(img topImage, columns []imgColumn, imgSpacing bool, normalColor, topNameColor, untaggedColor aec.ANSI) (out string) {
+	out += printNames(columns, img, topNameColor, untaggedColor)
+	out += printDetails(columns, normalColor, img.Details)
```

Comment:
Slightly on the fence on this, because it means that we'd now be buffering all output in a string, instead sending it to the writer 🤔

Reply:
I didn't think that was a big deal for this output, but I can change it back.

Reply:
Thinking about it, I think I can do something a little nicer with the other refactors I haven't pushed yet. One sec.